lib/openfeature/sdk/client.rb

```diff
@@ -1,18 +1,83 @@
+      end
+
+      def fetch_string_value(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def fetch_string_details(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def fetch_number_value(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def fetch_number_details(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def fetch_object_value(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def fetch_object_details(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
+
+      def evaluate_flag_details(flag_type:, flag_key:, default_value:, evaluation_context: nil)
+        case flag_type
+        when :boolean
+          @provider.fetch_boolean_value(flag_key: flag_key, default_value: default_value,
+                                        evaluation_context: evaluation_context)
+        when :number
+          @provider.fetch_number_value(flag_key: flag_key, default_value: default_value,
+                                       evaluation_context: evaluation_context)
+        when :string
+          @provider.fetch_string_value(flag_key: flag_key, default_value: default_value,
+                                       evaluation_context: evaluation_context)
+        when :object
+          @provider.fetch_object_value(flag_key: flag_key, default_value: default_value,
+                                       evaluation_context: evaluation_context)
+        else
+          StandardError.new("Unsupported flag_type")
```

Comment:
It's good to include relevant info in the error for someone reading it.

ArgumentError would be a better match for an error, since it is about the argument passed in.

And lastly, it needs to raise, not return 😅 

```suggestion
          raise ArgumentError.new("Unsupported flag_type: #{flag_type}")
```

Reply:
good call